### PR TITLE
Use `isinstance` instead of `if type() is ...`

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Replace `if type() is ...` with `if isinstance` as recommended in [PEP 8](https://peps.python.org/pep-0008/#programming-recommendations).

## Description of the changes
- Replace `if type() is ...` with `if isinstance` sentence.

### CODE REVIEW

**Comment by Alnusjaponica** [comment]
I am going to resolve `codecov` warning after (or when) https://github.com/optuna/optuna/issues/4959 is resolved.

File: optuna/integration/_lightgbm_tuner/optimize.py [hunk 1]
```diff
@@ -116,27 +118,27 @@ def _get_booster_best_score(self, booster: "lgb.Booster") -> float:
...
-            eval_at = [1, 2, 3, 4, 5]
+            or [1, 2, 3, 4, 5]
+        )
 
         # Optuna can handle only a single metric. Choose first one.
-        if type(eval_at) in [list, tuple]:
-            return "{}@{}".format(metric, eval_at[0])
-        if type(eval_at) is int:
-            return "{}@{}".format(metric, eval_at)
+        if isinstance(eval_at, (list, tuple)):
```
Comment (not522) [inline optuna/integration/_lightgbm_tuner/optimize.py:135]:
```suggestion
        if isinstance(eval_at, Sequence):
```

**Review by not522 — COMMENTED** [review]
Thank you, it's almost LGTM. Could you check my suggested change?

File: optuna/integration/_lightgbm_tuner/optimize.py (hunk 1 above)
Comment (not522) [reply optuna/integration/_lightgbm_tuner/optimize.py:135]: We discussed it offline and understood that it should not include string.

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**